test/test_irods_http_api.py

```diff
@@ -1825,6 +1825,332 @@ def test_non_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+
+        # Indicates whether the parallel_write_shutdown operation needs to be called.
+        # This is used for clean up when errors occur.
+        invoke_parallel_write_shutdown = False
+
+        try:
+            # Tell the server we're about to do a parallel write.
+            data_object = os.path.join('/', self.zone_name, 'home', self.rodsuser_username, 'parallel_write_header_based_0.txt')
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'parallel_write_init',
+                'lpath': data_object,
+                'stream-count': 3
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            parallel_write_handle = result['parallel_write_handle']
+
+            # Make sure the parallel_write_shutdown operation is invoked now that we
+            # have a parallel-write handle.
+            invoke_parallel_write_shutdown = True
+
+            # Write to the data object using the parallel write handle.
+            futures = []
+            with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
+                for e in enumerate(['A', 'B', 'C']):
+                    count = 10
+                    futures.append(executor.submit(self.execute_write_using_header_based_form, **{
+                        'bearer_token': self.rodsuser_bearer_token,
+                        'fields': {
+                            'op': 'write',
+                            'parallel-write-handle': parallel_write_handle,
+                            'offset': e[0] * count,
+                            'stream-index': e[0]
+                        },
+                        'bytes': e[1] * count
+                    }))
+
+                for f in concurrent.futures.as_completed(futures):
+                    result = f.result()
+                    self.logger.debug(result)
+                    self.assertEqual(result['irods_response']['status_code'], 0)
+
+            # End the parallel write.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'parallel_write_shutdown',
+                'parallel-write-handle': parallel_write_handle
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Read the contents of the data object and show it contains exactly what we expect.
+            r = requests.get(self.url_endpoint, headers=headers, params={
+                'op': 'read',
+                'lpath': data_object,
+                'count': 30
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.content.decode('utf-8'), 'A' * 10 + 'B' * 10 + 'C' * 10)
```

Comment:
Why are we not using the checksum in this test? Just curious.

Reply:
This may be from a time when the checksum operation wasn't quite there yet. Or, perhaps the test which this was derived from was attempting to prove the "read" operation worked as intended.